pyclam/manifold.py

```diff
@@ -280,6 +283,7 @@ def partition(self, *criterion) -> Iterable['Cluster']:
         2 children are produced if the cluster can be split, otherwise 1 child is produced.
         """
         if not all((
+                True if self.depth == 0 else self.name[-1] != '0',
```

Comment:
this reads weirdly, can it be done in two separate boolean clauses

Reply:
Checking just self.name[-1] == '0' breaks at the root level.
I would rather just punt this until we find what is really wrong with partition that it produced the empty list that I check for in line 329

Reply:
This is kinda equivalent to keeping a self.is_leaf flag